### DISCUSSION TITLE
Scores for transcriptions

### DISCUSSION THREAD

**Discussion by AllanBragaGit** [NVIDIA-NeMo/NeMo#4434] [Q&A] [answered]
At word level, i can extract the timestamps, how can i also extract a confidence level or a score from this word transcription?

i was following this tutorial: https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Offline_ASR.ipynb

**Comment by titu1994** [comment, marked as answer]
You could compute it yourself, with a somewhat new API that returns word and character timestamps. However scores themselves will require a bit more effort, you'll need to slice off the logprobs and average over the labels per word. 

For further information you can take a look at the main branch and use https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/metrics/wer_bpe.py#L27 

You can change the decoding strategy by using 
https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/models/ctc_models.py#L294 and use the normal transcribe() method to get back a Hypothesis object with all the values you need inside it https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/models/ctc_models.py#L114

After that, calculating stamps is practically using the output word timestamps, selecting the slice of the logprobs and then taking average over the selected non blank tokens as confidence measure.

**Reply by AllanBragaGit** [reply]
works perfectly, thanks for being so helpful!

**Comment by titu1994** [comment]
This feature is relative new, we'll document it in future releases when it becomes more stable. This can be done for any CTC model, it's not restricted to bpe models.
